app/models/variant_detail.rb

```diff
@@ -1,14 +1,16 @@
-class VariantDetail
-  attr_reader :split, :variant, :weight, :assignment_count
-
-  def initialize(split, variant)
-    @split = split
-    @variant = variant
-    @weight = split.variant_weight(variant)
-    @assignment_count = split.assignment_count_for_variant(variant)
+class VariantDetail < ActiveRecord::Base
+  belongs_to :split
+
+  validate :variant_must_exist, on: :create
```

Comment:
yeah I guess it can always run, I just didn't think there was a reason to prevent a variant that used to exist from having its metadata edited. but like we talked about before, that would only happen if someone was breaking the rules.